pkgs/dart_services/lib/src/compiler.dart

```diff
@@ -240,6 +240,67 @@ class Compiler {
     }
   }
 
+  /// Compile the given source file and return the resulting
+  /// [FlutterBuildResults].
+  Future<FlutterBuildResults> flutterBuild(String source) async {
+    final unsupportedImports = getUnsupportedImports(getAllImportsFor(source));
+    if (unsupportedImports.isNotEmpty) {
+      final message =
+          unsupportedImports.map((import) => import.uri.stringValue).join('\n');
+      return FlutterBuildResults.failed(message);
+    }
+
+    // TODO: Recycle this project directory.
+    final tempDir = await Directory.systemTemp.createTemp('dartpad');
+
+    try {
+      await copyPath(_projectTemplates.flutterPath, tempDir.path);
+
+      // Update lib/main.dart.
+      final sourceFile = File(path.join(tempDir.path, 'lib', kMainDart));
+      sourceFile.parent.createSync();
+      sourceFile.writeAsStringSync(source);
+
+      final arguments = <String>[
+        'build',
+        'web',
+
+        // This disables minification.
+        '--dart2js-optimization=O1',
+
+        // With the web renderer, we don't need to load other (skiawasm) resources.
```

Comment:
@kevmoo has asked that we look into instead using the skiawasm backend (likely loaded from a CDN). I'll add a todo.